### PR TITLE
Refactor GitopsService reconciler

### PR DESCRIPTION
**Pull request by jgwest**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

/kind code-refactoring

**What does this PR do / why we need it**:

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #? N/A

**Test acceptance criteria**:

* [ ] Unit Test
* [ ] E2E Test

**How to test changes / Special notes to the reviewer**:

This PR is a very simple refactor of the GitopsService reconciler.

This PR: 
- Fixes typo: `depracated`
- Splits the existing reconciler into two separate methods:
    -  `reconcileDefaultArgoCDInstance(instance, reqLogger)`
    - `reconcileBackend(gitopsserviceNamespacedName, instance, reqLogger)`
- Some whitespace changes 
    - I'm using curly braces to separate the scope for each section in `reconcileBackend`, let me know if you don't like this stylistically and I'll remove them, but personally I find them very helpful
        - Looks like [this in the code](https://github.com/redhat-developer/gitops-operator/blob/6d036693c2c3084bb0896c2a1feb5c39f7b22340/pkg/controller/gitopsservice/gitopsservice_controller.go#L317)
        - It makes it easy to see what each section of the function is doing, it limits the scope of variables, and prevents the need to pollute the code with  new functions for each section.

### CODE REVIEW

File: pkg/controller/gitopsservice/gitopsservice_controller.go [hunk 1]
```diff
@@ -287,168 +307,191 @@ func (r *ReconcileGitopsService) Reconcile(request reconcile.Request) (reconcile
...
+			} else {
 				return reconcile.Result{}, err
 			}
-		} else {
-			return reconcile.Result{}, err
 		}
 	}
 
-	return r.reconcileCLIServer(instance, request)
+	return reconcile.Result{}, nil
```
Comment (jgwest) [inline pkg/controller/gitopsservice/gitopsservice_controller.go:494]: FYI the call to reconcileCLIServer is still here, it's just been moved to the parent.

File: pkg/controller/gitopsservice/gitopsservice_controller.go [hunk 2]
```diff
@@ -287,168 +307,191 @@ func (r *ReconcileGitopsService) Reconcile(request reconcile.Request) (reconcile
...
-			reqLogger.Info("Creating a new Cluster Role", "Name", clusterRoleObj.Name)
-			err = r.client.Create(context.TODO(), clusterRoleObj)
+		existingClusterRole := &rbacv1.ClusterRole{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: clusterRoleObj.Name}, existingClusterRole)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				reqLogger.Info("Creating a new Cluster Role", "Name", clusterRoleObj.Name)
+				err = r.client.Create(context.TODO(), clusterRoleObj)
+				if err != nil {
+					return reconcile.Result{}, err
```
Comment (jgwest) [inline pkg/controller/gitopsservice/gitopsservice_controller.go:356]: FYI I made no change to this block of besides whitespace, not sure why GitHub is considering it separate...

**Comment by jgwest** [comment]
This is ready for review.

**Comment by iam-veeramalla** [comment]
Hi @jgwest , Just curious if there is any motivation behind the refactoring ? or is it just to make the code more readable ?

**Comment by jgwest** [comment]
Hey @iam-veeramalla , this is the first part of the fix for https://issues.redhat.com/projects/GITOPS/issues/GITOPS-997

**Comment by iam-veeramalla** [comment]
> Hey @iam-veeramalla , this is the first part of the fix for https://issues.redhat.com/projects/GITOPS/issues/GITOPS-997



> Hey @iam-veeramalla , this is the first part of the fix for https://issues.redhat.com/projects/GITOPS/issues/GITOPS-997

Thanks @jgwest

**Review by chetan-rns — APPROVED** [review]
Thanks!

/lgtm

**Pull request merged**